lib/application/index.js

```diff
@@ -4,6 +4,9 @@
  * configuration and rendering.
  */
 
+// Local
+var Promise = require('../polyfills/promise');
```

Comment:
Idk. That seems pretty opinionated for a library that could be used by different consumers with different browser support levels.